### PR TITLE
Documentation for ablator/main/mp.py

### PR DESCRIPTION
**Pull request by hieuchi911**
Documentation for `ablator/main/mp.py`

### CODE REVIEW

File: ablator/main/mp.py [hunk 1]
```diff
@@ -108,7 +168,47 @@ def handle_exception(e):
 
 
 class ParallelTrainer(ProtoTrainer):
+    """
+    A class for parallelizing training of models of different configurations with ray.
+    Performance of these models (metrics) are also logged in order for optuna to tune hyperparameters.
```
Comment (hjzccc) [inline ablator/main/mp.py]: maybe logging and hyperparameter optimizing are two different processes?

**Review by MrLYG — APPROVED** [review]
looks good for me.

File: ablator/main/mp.py [hunk 2]
```diff
@@ -319,6 +442,26 @@ def launch(
         auxilary_modules: list[tys.ModuleType] | None = None,
         ray_head_address: str | None = "auto",
     ):
+        """
+        Set up and launch the parallel training and tuning process. This includes:
+        prepare ray cluster for running optuna trials to tune hyperparameters; if available,
+        synchronize GCP buckets to working directory defined in runtime configuration;
+        initialize optuna trials and add them to optuna storage and experiment state
```
Comment (hjzccc) [inline ablator/main/mp.py:451]: should we add some function calls or class names to make it more clear,like experiment state is for storing and managing optuna samples,and sync_up is for synchronization, so users can refer to these files for more details

File: ablator/main/mp.py [hunk 3]
```diff
@@ -56,6 +87,35 @@ def train_main_remote(
     fault_tollerant: bool = True,
     crash_exceptions_types: list[type] | None = None,
 ) -> tuple[ParallelConfig, dict[str, float] | None, TrialState]:
+    """
+    The trial job that will be executed remotely at a ray node. This is where model training happens.
+    In addition, experiment directory will be synchronized to the GCP and remote nodes.
```
Comment (hjzccc) [inline ablator/main/mp.py]: maybe mention the function where the sychronization happens is clearer for users to find out how it works

**Review by hjzccc — APPROVED** [review]

**Review by fostiropoulos — APPROVED** [review]

**Pull request merged**